### PR TITLE
Remove function clause from `Internal`

### LINKED ISSUE #2363: Remove function clause from `Internal`
Function clauses are syntax sugar that do not belong in `Internal`. We should use lambdas instead.

### PR DESCRIPTION
**Pull request by janmasrovira**
- closes #2363

### CODE REVIEW

File: src/Juvix/Compiler/Internal/Extra/Base.hs [hunk 1]
```diff
@@ -158,24 +158,18 @@ subsHoles s = over leafExpressions helper
...
-  leafExpressions f c = do
-    _clauseBody <- leafExpressions f (c ^. clauseBody)
-    _clausePatterns <- traverse (leafExpressions f) (c ^. clausePatterns)
-    pure FunctionClause {_clauseName = c ^. clauseName, ..}
-
 instance HasExpressions Example where
   leafExpressions f = traverseOf exampleExpression (leafExpressions f)
 
 instance HasExpressions FunctionDef where
   -- leafExpressions f (FunctionDef name ty clauses bi) = do
```
Comment (lukaszcz) [inline src/Juvix/Compiler/Internal/Extra/Base.hs:165]: Useless comment (but it was here before)

File: src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/ArityChecking/Checker.hs [hunk 2]
```diff
@@ -132,11 +131,12 @@ checkFunctionDef ::
 checkFunctionDef FunctionDef {..} = do
   let arity = typeArity _funDefType
   _funDefType' <- withEmptyLocalVars (checkType _funDefType)
-  _funDefClauses' <- mapM (checkFunctionClause arity) _funDefClauses
+  -- _funDefClauses' <- mapM (checkFunctionClause arity) _funDefClauses
```
Comment (lukaszcz) [inline src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/ArityChecking/Checker.hs:134]: Remove the comment

File: src/Juvix/Compiler/Internal/Extra/Base.hs (hunk 1 above)
Comment (janmasrovira) [reply src/Juvix/Compiler/Internal/Extra/Base.hs]: I've removed it

File: src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/ArityChecking/Checker.hs (hunk 2 above)
Comment (janmasrovira) [reply src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/ArityChecking/Checker.hs]: done

**Review by lukaszcz — APPROVED** [review]

**Review by lukaszcz — APPROVED** [review]

**Pull request merged**